market/binance.go

```
@@ -19,35 +19,36 @@ func NewBinanceExchange(apiKey, secretKey string) *BinanceExchange {
 }
 
 func (b *BinanceExchange) Symbols() ([]string, error) {
+	start := time.Now()
 	client := binance.NewClient(b.apiKey, b.secretKey)
 	ei, err := client.NewExchangeInfoService().Do(context.Background())
 	if err != nil {
 		return nil, err
 	}
-	symbols := make([]string, 0)
+	symbols := make([]string, 0) // we don't know yet how many symbols we will be tradeable
 	for _, s := range ei.Symbols {
 		if s.IsMarginTradingAllowed && s.IsSpotTradingAllowed {
 			symbols = append(symbols, s.Symbol)
 		}
 
 	}
+
+	elapsed := time.Since(start)
+	log.Println("Found", len(symbols), "Tokens in", elapsed)
+	log.Println("Symbols:", symbols)
 	return symbols, nil
 }
 
 func (b *BinanceExchange) ProduceAllPrice(errorHandler ErrorHandler) *BinanceProducer {
-	start := time.Now()
 	tokens, err := b.Symbols()
 	if err != nil {
 		errorHandler(err)
```

You should return the error here, doesn't make sense to go forward if the tokens slice is empty.
Plus I prefer to wrap the errors, either with just extra information about the actual functionality or with a stack trace too.

The same goes for all the `errorHandler` solutions, you do not actually handle the errors, when you should, especially because these are sync calls.